HyperVExtension/src/HyperVExtension/Services/IDownloaderService.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using HyperVExtension.Models.VirtualMachineCreation;
+
+namespace HyperVExtension.Services;
+
+/// <summary>
+/// Interface for a service that can download files from the web.
+/// </summary>
+public interface IDownloaderService
+{
+    /// <summary>
+    /// Starts file a download operation asynchronously from the web.
+    /// </summary>
+    /// <param name="progressProvider">The provider who progress should be reported back to</param>
+    /// <param name="sourceWebUri">The web uri that points to the location of the file</param>
+    /// <param name="destinationFile">The file path that the downloaded file should be downloaded into</param>
+    /// <param name="cancellationToken">A token that can allow the operation to be cancelled while it is running</param>
+    /// <returns>A wrapper for the DownloadOperation which contains the status of the operation</returns>
```

Comment:
Do you mean that "wrapper" is Task itself? It's inconsistent with other <return> comments here.
For Task<string> below it says, "String content returned by web server", the result of the operation, not Task.

Reply:
Sorry that was an old comment I forgot to update this thanks